website/content/platform/development/contributing/how-to/add_data_to_existing_endpoint.md

```diff
@@ -0,0 +1,860 @@
+    cloud-based APIs, Excel, and Google Sheets. """,
+    credentials=["api_key"],
+    fetcher_dict={
+        "EquityHistorical": AVEquityHistoricalFetcher,
+        "HistoricalEps": AVHistoricalEpsFetcher,
+    },
+)
+```
+
+Step 5 is complete.
+  - [X] Map the new provider model to the router.
+
+## Rebuild Static Assets
+
+When modifying router components or model definitions, the Python interface needs to be rebuilt before use. Open a terminal, with the `obb` environment active, start a new Python session, and enter:
+
+```python
+import openbb
+
+openbb.build()
+
+exit()
+```
+
+:::note
+
+If changes are only made to the static methods within the Fetcher, rebuilding is not required. Restart the Python interpreter to apply the edits.
+
+:::
+
+Step 6 is done.
+  - [X] Rebuild the Python interface and static assets.
+
+We can now run the function and test our work.
+
+```python
+from openbb import obb
+
+obb.equity.fundamental.historical_eps(
+    symbol=["IBM","GOOG","AAPL","MSFT"],
+    period="quarter",
+    provider="alpha_vantage",
+    limit=1
+).to_df()
+```
+
+| date       | symbol   |   eps_actual |   eps_estimated |   surprise |   surprise_percent | reported_date   |
+|:-----------|:---------|-------------:|----------------:|-----------:|-------------------:|:----------------|
+| 2023-12-31 | GOOG     |         1.64 |            1.59 |       0.05 |           0.031447 | 2024-01-30      |
+| 2023-12-31 | AAPL     |         2.18 |            2.1  |       0.08 |           0.038095 | 2024-02-01      |
+| 2023-12-31 | MSFT     |         2.93 |            2.78 |       0.15 |           0.053957 | 2024-01-30      |
+| 2023-12-31 | IBM      |         3.87 |            3.78 |       0.09 |           0.02381  | 2024-01-24      |
+
+Checking the `annual` setting:
+
+```python
+obb.equity.fundamental.historical_eps(
+    symbol="AAPL,
+    period="annual",
+    provider="alpha_vantage",
+    limit=4
+).to_df()
+```
+
+| date       | symbol   |   eps_actual |
+|:-----------|:---------|-------------:|
+| 2021-09-30 | AAPL     |         5.62 |
+| 2022-09-30 | AAPL     |         6.11 |
+| 2023-09-30 | AAPL     |         6.12 |
+| 2023-12-31 | AAPL     |         2.18 |
+
+We can see that the most recent `annual` data point only represent the first quarter of Apple's fiscal year, and this is something to keep in mind while working with the data.
+
+To check that the warning is being transmitted, enter a bad symbol in the list.
+
+```python
+obb.equity.fundamental.historical_eps(symbol="AAPL,BAD_SYMBOL", provider="alpha_vantage").warnings
+```
+
+```console
+[Warning_(category='UserWarning', message='Symbol Error: No data found for BAD_SYMBOL')]
+```
+
+With confidence that the endpoint is working as expected, let's move on to unit and integration tests.
+
+## Add Tests
+
+Adding tests doesn't take a lot of effort. In most cases, copying and pasting from an existing one will do the job.
```

Comment:
There is also a command to run that will generate the test for the user. Let's share it.